AjobK/Athena
# 270 cookies request

closes #271 

<br />

 **What is it supposed to do**  
Uses the PopUp component to request the user to accept cookies. Closing the popup stores the current date in localStorage, which is used each time the application is entered for verifying the cookies (after 1 week the request should appear again). 

A GlobalNotification component was added to the root of the application. This component has a listener for the new notification store, which allows a very easy way to create popups anywhere in the application:
![image](https://user-images.githubusercontent.com/14930277/132095403-a6b0eb34-f78b-4399-a32c-0ab4c074f0f7.png)

_NOTE: At the moment this only works for popups without custom children, so only the title, description and buttons can be modified. Shouldn't be a difficult fix, depending on the ways I can store this data in a mobx-state-tree store. The GlobalNotification is located in the AppRouter component instead of the App component as I couldn't get it working in App (seems like App isn't being used at all?)._

![image](https://user-images.githubusercontent.com/14930277/132095291-18049b50-199d-4c27-8f88-d60ab225983a.png)


<br />

 **How can we test your branch**  
Launch the application. The popup above should appear. Refreshing the page without accepting the cookies (clicking on the 'x' or outside of the popup) should show the popup again, until it is accepted.
![image](https://user-images.githubusercontent.com/14930277/132095623-db2fc511-34e7-44fe-b0d5-99d4e4b9ee2a.png)

You could try deleting and/or modifying the "cookiesAcceptedAt" value in the local storage and refresh the page (modifying the value to an invalid date should make it invalid). You could try changing the date in the localStorage to another (valid) date string that is more than a week before the current date, to see if the popup appears when the cookies aren't valid anymore.

As for the general global notifications, you could try adding one to any component in the application by calling:
`this.props.store.notification.setContent(popUpData.messages.networkError)`
(Make sure to import the popUpData file)

<br />

**1. General checklist**

<br />

**2. If relevant, front-end checklist**

<br />

**3. If relevant, back-end checklist**

<br />

**4. If relevant, test these browsers**
